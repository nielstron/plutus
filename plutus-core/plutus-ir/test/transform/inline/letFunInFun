{- 
Single occurrence of a function (g) that calls another let binding.
The unconditional inlining optimization should inline g and idFun because they only occur once.

let 
  idFun :: Integer -> Integer
  idFun x = x
  g :: Integer -> (Integer -> Integer)
  g y = idFun
in g 3
 -}
(let
    (nonrec)
    (termbind
      (strict)
      (vardecl idFun (fun (con integer) (con integer)))
      (lam x (con integer) x)
    )
    (termbind
      (strict)
      (vardecl g (fun (con integer) (fun (con integer) (con integer))))
<<<<<<< HEAD
      (lam y (con integer) idFun)
=======
      (lam eta (con integer) idFun)
>>>>>>> de7c772ed (f)
    )
    [ g (con integer 3) ]
)