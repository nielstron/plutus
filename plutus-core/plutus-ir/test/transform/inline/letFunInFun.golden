<<<<<<< HEAD
<<<<<<< HEAD
[ (lam y (con integer) (lam x (con integer) x)) (con integer 3) ]
=======
(lam x (con integer) x)
>>>>>>> f09f6ba6a (Add letFunInFun test.)
=======
[ (lam eta (con integer) (lam x (con integer) x)) (con integer 3) ]
>>>>>>> de7c772ed (f)
